server/segment_rpc.go

```diff
@@ -484,15 +497,42 @@ func genSegCreds(sess *BroadcastSession, seg *stream.HLSSegment) (string, error)
 	return base64.StdEncoding.EncodeToString(data), nil
 }
 
-func newBalanceUpdate(sess *BroadcastSession) (*BalanceUpdate, error) {
+func estimateFee(seg *stream.HLSSegment, profiles []ffmpeg.VideoProfile, priceInfo *big.Rat) (*big.Rat, error) {
+	if priceInfo == nil {
+		return nil, nil
+	}
+
+	// TODO: Estimate the number of input pixels
+	// Estimate the number of output pixels
+	var outPixels int64
+	for _, p := range profiles {
+		w, h, err := ffmpeg.VideoProfileResolution(p)
+		if err != nil {
+			return nil, err
+		}
+
+		// Take the ceiling of the duration to always overestimate
+		outPixels += int64(w*h) * int64(p.Framerate) * int64(math.Ceil(seg.Duration))
+	}
+
+	// feeEstimate = pixels * pixelEstimateMultiplier * priceInfo
+	fee := new(big.Rat).SetInt64(outPixels)
+	// Multiply pixels by pixelEstimateMultiplier to ensure that we never underpay
```

Comment:
Since we overestimate the number output pixels by rounding up the segment duration do we still need `pixelEstimateMultiplier` wouldn't either suffice?

Reply:
I added `pixelEstimateMultiplier` just to be safe in the scenario where the segment duration is already an integer value (i.e. 2.0). I believe it is possible for the # of frames in the output to not always exactly equal FPS * duration - see [this example](https://github.com/livepeer/lpms/blob/master/ffmpeg/ffmpeg_test.go#L520) where we expect 125 frames for a 1 second segment at 123 FPS